src/org/sosy_lab/java_smt/solvers/cvc5/CVC5InterpolatingProver.java

```diff
@@ -0,0 +1,368 @@
+    try { // Interpolate every Interpolation Pair
+      for (int i = 0; i < interpolationPairs.size(); i++) {
+        itps.add(
+            getCVC5Interpolation(
+                interpolationPairs.get(i).get(0), interpolationPairs.get(i).get(1)));
+      }
+    } catch (UnsupportedOperationException e) {
+      if (e.getMessage() != null) {
+        throw new SolverException(e.getMessage(), e);
+      } else {
+        throw e;
+      }
+    }
+
+    final List<BooleanFormula> result = new ArrayList<>();
+    for (Term itp : itps) {
+      result.add(creator.encapsulateBoolean(itp));
+    }
+    assert result.size() == pStartOfSubTree.length - 1;
+    return result;
+  }
+
+  /**
+   * Interpolates a Tuple of Interpolants according to Craig-Interpolation using CVC5-Interpolation.
+   *
+   * <p>CVC5's getInterpolant: There is a Model, such that the Interpolant I, with A -> I = True
+   * (1CVC5) and I -> B = True (2CVC5).
+   *
+   * <p>Craig Interpolation: There is a Model, such that the Interpolant psi, with phi- -> psi =
+   * True (1Craig), not(psi /\ phi+) = True (2Craig).
+   *
+   * <p>With A current set of assertions and B Formulas to interpolate.
+   *
+   * <p>CVC5 -> Craig Interpolation:
+   *
+   * <p>(1CVC5) <=> (1Craig) due to subst of A with phi- and reflexivity.
+   *
+   * <p>(2CVC5) <=> I -> B = True <=> (not I) \/ B = True <=> not (I /\ (not B)) = True <=> (2Craig)
+   * due to subst of (not B) with phi+ and reflexivity.
+   *
+   * <p>Hence, CVC5's Interpolation produces an equivalent Interpolation to Craig Interpolation, if
+   * B is negated during CVC5 Interpolation.
+   *
+   * @param assertedInterpols Asserted Formulas
+   * @param addedInterpols Formulas to Interpolate
+   * @return Interpolation of the Interpolation Pair following the definition of
+   *     Craig-Interpolation.
+   */
+  private Term getCVC5Interpolation(
+      ArrayList<Collection<Term>> assertedInterpols, ArrayList<Collection<Term>> addedInterpols) {
+
+    // Respect Asserted Formulas not in the Interpolation Pairs
+
+    ArrayList<Collection<Term>> combinedInterpols = new ArrayList<>();
+    combinedInterpols.addAll(assertedInterpols);
+    combinedInterpols.addAll(addedInterpols);
+
+    // checks, that no Assertions in the Assertion stack are left out. Can happen in Tree
+    // Interpolation
+    Collection<Term> extraAssertions = getAssertedTermsNotInCollection(combinedInterpols);
+    Term extraAssert = new Term();
+
+    Solver interpolationSolver = new Solver(); // Uses a separate Solver instance to leave to
+    // original solver-context unmodified
+
+    setSolverOptions(seed, solverOptions, interpolationSolver);
+
+    if (extraAssertions.isEmpty()) {
+      extraAssert = interpolationSolver.mkTrue();
+    } else {
+      extraAssert = buildConjunctionOfFormulas(extraAssertions, interpolationSolver);
+    }
+
+    Term A = buildConjunctionOfFormulasOverArray(assertedInterpols, interpolationSolver);
```

Comment:
You could just build `extraAssert` after this line and add `extraAssert` to a with an conjunction.